### PR TITLE
[openal-soft] update to 1.23.0

### PR DESCRIPTION
**Pull request by quyykk**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

The removed patch was upstreamed.

### CODE REVIEW

File: ports/openal-soft/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "openal-soft",
-  "version-semver": "1.22.2",
-  "port-version": 5,
+  "version-semver": "1.23.0",
```
Comment (JonLiu1993) [inline ports/openal-soft/vcpkg.json:3]: @quyykk, Thanks for your PR, could you please confirm whether the library officially adheres to semantic versioning or not (breaking changes -> bump major version, non-breaking changes -> bump minor version, non-breaking bug fixes -> bump patch version)?

Comment (quyykk) [reply ports/openal-soft/vcpkg.json:3]: I can't find anything that suggests it uses semantic versioning.... Should I remove it?

Comment (JonLiu1993) [reply ports/openal-soft/vcpkg.json:3]: Please change "version-semver" to "version".

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature `pipewire` tested successfully in the x64-linux triplet.

**Pull request merged**